### ISSUE TITLE
Trait implementations for generated types

### ISSUE THREAD

**Issue by adeschamps** [adeschamps/lcm-rust#29]
Right now, `lcm-gen` adds `#[derive(Debug, Message)]` to each generated struct. Which other traits should be implemented by default?

Although it's trivial, I don't think it makes sense in general to implement traits related to comparisons (`PartialEq`, `PartialOrd`, `Hash`, etc.)

`Copy` and `Clone` seem like obvious choices.

Serde traits could be implemented via a feature flag.

**Comment by adeschamps** [comment]
It might make sense to add an option to `lcm-gen` so that the user can list additional traits to implement.

**Comment by neachdainn** [comment]
> It might make sense to add an option to lcm-gen so that the user can list additional traits to implement.

I thought I saw something like that listed in the options last time I ran the (new) `lcm-gen`? I think that's a very good idea.

> `Copy` and `Clone` seem like obvious choices.

I think `Clone` is a good choice. `Copy` probably can't be derived since you would have to know the whole tree to see if a `String` ever appears in it.

> Serde traits could be implemented via a feature flag.

I don't think this will be necessary if we add the option to `lcm-gen`.

> I don't think it makes sense in general to implement traits related to comparisons 

If we don't have the `lcm-gen` option, I think `PartialEq` is something we need to include. With the option, I think `Clone`, `Debug`, and `Message` are sufficient.
